applications/GeoMechanicsApplication/tests/cpp_tests/test_thermal_filter_law.cpp

```diff
@@ -0,0 +1,48 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//
+
+#include "custom_constitutive/thermal_filter_law.h"
+#include "geo_mechanics_application.h"
+#include "includes/ublas_interface.h"
+#include "testing/testing.h"
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(CalculateThermalFilterLawMatrix, KratosGeoMechanicsFastSuite)
+{
+    Model current_model;
+    auto& r_model_part = current_model.CreateModelPart("ModelPart");
+
+    auto cond_prop = r_model_part.CreateNewProperties(0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_WATER, 1000.0);
+
+    const SizeType      dimension = 1;
+    GeoThermalFilterLaw geo_thermal_filter_law(dimension);
+    ProcessInfo         info;
+
+    const Matrix thermal_filter_matrix = geo_thermal_filter_law.CalculateThermalDispersionMatrix(*cond_prop, info);
+
+    Matrix expected_solution = ScalarMatrix(1, 1, 1000.0);
+
+    constexpr double tolerance{1.0e-6};
+
+    KRATOS_EXPECT_NEAR(thermal_filter_matrix(0, 0), expected_solution(0, 0), tolerance);
```

Comment:
I would change this check to a matrix-based one (which also checks the dimensions):
```suggestion
    KRATOS_EXPECT_MATRIX_NEAR(thermal_filter_matrix, expected_solution, tolerance);
```